code/modules/mob/mob.dm

```diff
@@ -230,7 +233,24 @@
 			del(narglow)
 
 /mob/proc/see_rift(var/obj/machinery/singularity/narsie/large/exit/R)
-	if((R.z == src.z) && (get_dist(R,src) <= (R.consume_range+10)) && !(R in view(src)))
+	if(istype(src.loc,/obj/mecha/))
```

Comment:
I'll try that once I'm done with the Preferences changes I'm working on right now